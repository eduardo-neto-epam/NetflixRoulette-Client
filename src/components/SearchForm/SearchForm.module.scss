.container {
    display: block;
    justify-content: flex-start;
    flex-wrap: nowrap;
    width: var(--content-max-width);
    margin: 0 auto;

    .input {
        width: 80%;
        margin-right: 40px;
        height: 58px;
        background-color: rgba(35, 35, 35, 0.4);
        background-image: linear-gradient(rgba(0,0,0,0.4), rgba(35,35,35,0.9));
        border: none;
        border-radius: 6px;
        color: #ffffff;
        font-family: var(--font-main);
        font-size: 24px;
        font-weight: 400;
        letter-spacing: 1px;
        padding: 0 20px;
    
        &:after{
            content: '';
            position: absolute;
            top: 2px;
            left: 2px;
            width: calc(100% - 4px);
            height: 50%;
            background: linear-gradient(rgba(255,255,255,0.8), rgba(255,255,255,0.2));
        }
    
        &::placeholder {
            color: #424242;
            letter-spacing: 1.25px;
        }
    }
}

