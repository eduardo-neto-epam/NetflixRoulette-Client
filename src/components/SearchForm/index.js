import React, { useState } from 'react'
import styles from './SearchForm.module.scss'
import classnames from 'classnames/bind'
import Button from '../Button'

let cx = classnames.bind(styles)

const SearchForm = () => {

    const [ value, setValue ] = useState('')

    function handleChange(e) {
        e.persist()
        setValue(e.target.value)
    }

    function handleSubmit(e) {
        e.preventDefault()
        console.log('Sumbiting: ', value)
    }

    return (
        <div className={cx('container')}>
            <form onSubmit={handleSubmit}>
                <input
                    className={cx('input')}
                    value={value}
                    placeholder='What do you want to watch?'
                    onChange={handleChange}
                    required
                />
                <Button type='submit' size='lg'>SEARCH</Button>
            </form>
        </div>
    )
}

export default SearchForm;