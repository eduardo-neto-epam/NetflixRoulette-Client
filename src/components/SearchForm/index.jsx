import React, { useState } from 'react';
import classnames from 'classnames/bind';
import styles from './SearchForm.module.scss';
import Button from '../Button/index.jsx';

const cx = classnames.bind(styles);

const SearchForm = () => {
  const [value, setValue] = useState('');

  function handleChange(e) {
    e.persist();
    setValue(e.target.value);
  }

  function handleSubmit(e) {
    e.preventDefault();
    // eslint-disable-next-line no-console
    console.log('Sumbiting: ', value);
  }

  return (
    <div className={cx('container')}>
      <form onSubmit={handleSubmit}>
        <input
          className={cx('input')}
          value={value}
          placeholder="What do you want to watch?"
          onChange={handleChange}
          required
        />
        <Button type="submit" size="lg">SEARCH</Button>
      </form>
    </div>
  );
};

export default SearchForm;
