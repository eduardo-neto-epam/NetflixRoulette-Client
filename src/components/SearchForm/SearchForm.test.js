import renderer from 'react-test-renderer';
import React from 'react'
import SearchForm from './index'

test('SearchForm renders correctly', () => {
    const tree = renderer
        .create(<SearchForm />)
        .toJSON();
    expect(tree).toMatchSnapshot();
  });