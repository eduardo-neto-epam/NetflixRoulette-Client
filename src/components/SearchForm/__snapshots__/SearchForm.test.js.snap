// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm renders correctly 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <input
      className="input"
      onChange={[Function]}
      placeholder="What do you want to watch?"
      required={true}
      value=""
    />
    <button
      className="btn btn--lg btn--primary"
      onClick={[Function]}
      type="submit"
    >
      SEARCH
    </button>
  </form>
</div>
`;
